packages/@react-aria/focus/src/isElementVisible.ts

```diff
@@ -43,6 +43,8 @@ function isStyleVisible(element: Element) {
 function isAttributeVisible(element: Element, childElement?: Element) {
   return (
     !element.hasAttribute('hidden') &&
+    // Ignore HiddenSelect when tree walking.
+    !(element.hasAttribute('data-a11y-ignore') && element.getAttribute('aria-hidden') === 'true') &&
```

Comment:
won't this conflate hiding things from the axe plugin vs implementation?

Reply:
maybe, this came from Michael's pr. I'm on pto next week but you could ask him about it. Not sure why just checking aria-hidden without the data-a11y-ignore wouldn't work.

Reply:
It came from my PR: https://github.com/adobe/react-spectrum/pull/5878/commits/d160bac41a51c839858b0c4614fb95da5e799ffd#diff-e1bb593250b04d4c80095b5a414062150d86db252d886ee3a8040c7eb126425b

We can test with just skipping `aria-hidden`.

Reply:
Simply ignoring `!(element.getAttribute('aria-hidden') === 'true')` will break tree walking to Tab/Shift+Tab focus the next or previous element when a Picker or Menu is open, because all elements under a modal popover have `aria-hidden="true"`.

Reply:
Seems like we should introduce a new data attribute then? there may be other reasons for this data attribute being used, and it can be used by non-rsp projects on any component. It's very specific to AXE and other related tools

Reply:
Any suggestions? `data-ignore-tree-walking`?

Reply:
yeah, i think that's a good place to start the discussion, thank you!